crates/astria-config/src/lib.rs

```diff
@@ -0,0 +1,133 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{
+    parse_macro_input,
+    Item,
+};
+
+#[proc_macro_attribute]
+pub fn astria_config(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let item_ast = parse_macro_input!(item as Item);
+
+    let Item::Struct(ref item_struct) = item_ast else {
+        panic!("the astria_config proc macro can only be called on structs");
+    };
+
+    let struct_name = item_struct.ident.clone();
+
+    // Generates an vec of token-tree like structures of the atomicity we care about
+    let struct_fields = item_struct
+        .fields
+        .iter()
+        .map(|x| {
+            let field_ident = x.ident.clone().unwrap();
+            let field_ty = x.ty.clone();
+            let field_vis = x.vis.clone();
+
+            (field_ident, field_ty, field_vis)
+        })
+        .collect::<Vec<_>>();
+
+    // Check if log is a valid member of the config, otherwise panic
+    let true = struct_fields
+        .iter()
+        .map(|x| x.0.to_string().to_lowercase())
+        .collect::<Vec<_>>()
+        .contains(&"log".to_string())
+    else {
+        panic!("config struct does not contain a log element, which is necessary");
+    };
+
+    // Check if the macro itself has a valid attribute
+    let attr_str = attr.to_string().to_lowercase();
+    match attr_str.as_str() {
+        "conductor" | "composer" | "sequencer" | "sequencer_relayer" => {}
```

Comment:
I don't necessarily agree with this - I think this config setup only works for these services because we had tests for these services - if we add services in the future that make use of this macro, I think we should follow a process in which the config is set up vanilla -> and then migrated to this macro to see if it behaves the same and changing the macro as desired. Restricting is like this would be my preference.

Reply:
Also another point to note is that this only follows the config spec for services, not for user facing cli tools for instance. Making this agnostic would be confusing IMO

Reply:
Update: Spoke with @SuperFluffy briefly, also came up with an in-between solution that doesn't compromise readability, but also is more generalized than this implementation